lib/serviceDiscovery/helpers.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const os = require('os');
+const crypto = require('crypto');
+const memoizee = require('memoizee');
+
+const interfaces = os.networkInterfaces();
+const ifaceNames = Object.keys(interfaces);
+
+// choose the first valid ip as the one we will announce
+let announceIps;
+
+/**
+ * Builds a list of every announce-able ips on the server
+ *
+ * @returns {string} The ip list
+ * @todo Support from env variables?
+ */
+exports.getAnnounceIps = memoizee(function () {
+	if (announceIps) {
+		return announceIps;
+	}
+
+	announceIps = [];
+
+	for (var i = 0; i < ifaceNames.length; i++) {
+		var addresses = interfaces[ifaceNames[i]];
+
+		for (var j = 0; j < addresses.length; j++) {
+			var address = addresses[j];
+
+			// skip anything that doesn't interest us
+			if (address.internal) {
+				continue;
+			}
+
+			// found it, store it and break
```

Comment:
There's no `break`

Reply:
Think I remember now: initially I wanted to use the first announce-able IP, but I didn't trust that this would be consistent enough on all platform and ended up using **all** announced IPs instead. Comments are from that initial concept.